xapian-glib/xapian-database.cc

```diff
@@ -638,11 +663,11 @@ xapian_database_compact_to_path (XapianDatabase             *self,
 
   int real_flags = 0;
 
-  if (flags & XAPIAN_DB_COMPACT_NO_RENUMBER)
+  if ((flags & XAPIAN_DB_COMPACT_NO_RENUMBER) == XAPIAN_DB_COMPACT_NO_RENUMBER)
```

Comment:
This is not right; `Xapian::DBCOMPACT_*` values **are** a bitmask, and they are handled appropriately in the Xapian source code.